.github/workflows/ci-pipeline.yml

```
@@ -0,0 +1,36 @@
+name: Continuous Integration
+on:
+  push:
+    paths-ignore:
+      - '**/README.md'
+
+  pull_request:
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - run: docker build --target test --tag todo_app:test .
+    - run: docker run --platform linux/amd64 --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app todo_app:test todo_app/tests
+    - run: docker run --platform linux/amd64 --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e TRELLO_API_TOKEN=${{ secrets.TRELLO_API_TOKEN }} -e ID_ORGANIZATION=${{ secrets.ID_ORGANIZATION }} -e SECRET_KEY=${{ secrets.SECRET_KEY }} todo_app:test todo_app/tests_e2e
```

Preference rather than strong recommendation, but we could potentially break this up a little to improve* readability:
* First we take advantage of the fact that Docker can make an assumption if no value is passed for an environment variable, that it should match the host environment (i.e. `-e TRELLO_API_KEY` is equivalent to `-e TRELLO_API_KEY=$TRELLO_API_KEY`)
* Then we convert the secrets to environment variables separately:
```suggestion
    - run: docker run --platform linux/amd64 -e TRELLO_API_KEY -e TRELLO_API_TOKEN -e ID_ORGANIZATION -e SECRET_KEY todo_app:test todo_app/tests_e2e
      env:
          TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }}
          TRELLO_API_TOKEN=${{ secrets.TRELLO_API_TOKEN }}
          ID_ORGANIZATION=${{ secrets.ID_ORGANIZATION }}
          SECRET_KEY=${{ secrets.SECRET_KEY }}
```

*Whether this helps is definitely a matter of opinion - I can see an argument that all we've done is add extra steps!